capstone/src/main/webapp/script.js

```diff
@@ -53,3 +85,67 @@ function createAnyElement(tag, text) {
   textElement.innerHTML = text;
   return textElement;
 }
+
+/** Loads the Google Charts API */
+function loadChartsApi(toxicityData) {
+  google.charts.load('current', {'packages':['corechart']});
+  google.charts.setOnLoadCallback(function() {drawBarChart(toxicityData);}); 
+}
+
+/** Draws a Google BarChart from a Perspective JSON. */
+function drawBarChart(toxicityData) {
+  document.getElementById('chart-container').innerHTML = '';
+  const data = google.visualization.arrayToDataTable([[{label: 'Attribute'}, {label: 'Score', type: 'number'}, {role: "style"}]]);
+
+  Object.keys(toxicityData.attributeScores).forEach((attribute) => {
+    var color = '#6B8E23'; // Green
+    const score = toxicityData.attributeScores[attribute].summaryScore.value;
+    if (score >= 0.8) {
+      color = '#DC143C'; // Red
+    } else if (score >= 0.2) {
+      color = '#ffd800'; // Yellow
+    }
+    data.addRow([attribute, score, color]);
+  });
+
+  data.sort({column: 1, desc: false});
+
+  const options = {
+    title: 'Perspective Feedback',
+    bars: 'horizontal',
+    height: 700,
+    legend: {position: "none"},
+    theme: 'material', 
+    hAxis: {viewWindow: {min: 0, max: 1}}
+  };
+
+  const chart = new google.visualization.BarChart(document.getElementById('chart-container'));
+  chart.draw(data, options);
+}
+
+/** Shows the avaiable attributes given a language selected on text analyzer page */
+function showAvailableAttributes() {
+  const langElement = document.getElementById('languageForAnalysis');
+  if (!langElement) {
+    return;
+  }
+  const lang = langElement.value;
+  const avaiableAttributesElement = document.getElementById('available-attributes');
+  avaiableAttributesElement.innerHTML = '';
+	
+  const attributes = ATTRIBUTES_BY_LANGUAGE[lang];
+  attributes.forEach(function(attribute) {
+    const checkbox = document.createElement('input');
+    checkbox.type = "checkbox";
+    checkbox.value = attribute;
+    checkbox.id = attribute + '-checkbox';
+    checkbox.checked = true;
+  
+    const label = document.createElement('label');
+    label.htmlFor = attribute + '-checkbox';
+    label.appendChild(document.createTextNode(attribute));
+  
+    avaiableAttributesElement.appendChild(checkbox);
+    avaiableAttributesElement.appendChild(label);
+  })
```

Comment:
Does this need a semicolon? I don't think we fully ended the function on line 137.

Reply:
Added one - I think it still compiled because semicolons are not always required in JavaScript